ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,38 @@ -(void)deleteObject{
     [self remove];
 }
 
+-(void)getExplorerUrl:(void (^)(NSString*))successcb onError:(void (^)(NSError*))errorcb {
+    NSLog(@"%@ getExplorerUrl",self.Id);
+    NSMutableString *path = [NSMutableString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@",
+                      self.report_id];
+    if ([self.test_name isEqualToString:@"web_connectivity"])
+        [path appendString:[NSString stringWithFormat:@"&input=%@",
+                            self.url_id.url]];
+    NSURL *url = [NSURL URLWithString:path];
+    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession]
+     dataTaskWithURL:url
+     completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+         if (error != nil) {
+             errorcb(error);
+             return;
+         }
+         NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+         if (error != nil) {
+             errorcb(error);
+             return;
+         }
+         NSArray *resultsArray = [dic objectForKey:@"results"];
+         //TODO symbolize somehow empty array
+         if ([resultsArray count] <= 0) {
+             errorcb([NSError errorWithDomain:@"io.ooni.api"
+                                         code:ERR_JSON_EMPTY
+                                     userInfo:@{NSLocalizedDescriptionKey:@"Error.JsonEmpty"
+                                                }]);
+             return;
+         }
+         successcb([[resultsArray objectAtIndex:0] objectForKey:@"measurement_url"]);
```

Comment:
I believe we should also check for the case where the `measurement_url` key does not exist.

Reply:
done

Reply:
I believe the error is changed and it's not necessarily that the JSON is empty. Maybe we should use a better error naming and characterisation such as, e.g., "Error.JsonMalformed"?